### PR TITLE
[SPARK-48522][BUILD] Update Stream Library to 2.9.8 and attach its NOTICE

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

Update Stream Library to 2.9.8 and attach its NOTICE

### Why are the changes needed?

update dep and notice file

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

passing ci

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

File: NOTICE-binary [hunk 1]
```diff
@@ -33,11 +33,12 @@ services.
 // Version 2.0, in this case for
 // ------------------------------------------------------------------
 
-Hive Beeline
```
Comment (yaooqinn) [inline NOTICE-binary:36]: `Hive Beeline` is duplicated

**Comment by yaooqinn** [comment]
cc @LuciferYang thanks

**Review by LuciferYang — APPROVED** [review]

**Comment by LuciferYang** [comment]
Unrelated to this PR, but the `stream-lib` has already been archived, perhaps we need to consider what alternatives there might be for the future.

**Comment by yaooqinn** [comment]
Yeah, it looks like we have a builtin HLL implementation to replace the one in steam-api

**Comment by LuciferYang** [comment]
Merged into master for Spark 4.0. Thanks @yaooqinn

**Pull request closed without merging**